src/main/scala/com/typesafe/sbt/packager/debian/DebianArtifactoryDeployPlugin.scala

```diff
@@ -0,0 +1,85 @@
+package com.typesafe.sbt
+package packager
+package debian
+
+import sbt._
+import sbt.Keys._
+import SbtNativePackager._
+import packager.Keys._
+
+import java.net.HttpURLConnection
+import javax.xml.bind.DatatypeConverter.printBase64Binary
+
+trait DebianArtifactoryDeployKeys {
+  val debianArtifactoryUrl = SettingKey[String]("debian-artifactory-url", " Url of Atrifactory server")
+  val debianArtifactoryRepo = SettingKey[String]("debian-artifactory-repo", "Name of Artifactory repository with deb layout to publish artifacts to")
+  val debianArtifactoryCredentials = SettingKey[Option[Credentials]]("debian-artifactory-credentials", "Credentials with permissions to publish to Artifactory server")
+
+  val debianArtifactoryPath = SettingKey[String]("debian-artifactory-path", "The path in repository where the package should be stored")
+
+  val debianArtifactoryDistribution = SettingKey[Seq[String]]("debian-artifactory-distribution", "The value to assign to the deb.distribution property used to specify the Debian package distribution")
+  val debianArtifactoryComponent = SettingKey[Seq[String]]("debian-artifactory-component", "The value to assign to the deb.component property used to specify the Debian package component name")
+  val debianArtifactoryArchitecture = SettingKey[Seq[String]]("debian-artifactory-architecture", "The value to assign to the deb.architecture property used to specify the Debian package architecture")
+
+  val debianArtifactoryTargetPath = TaskKey[String]("debian-artifactory-apt-target-path", "Construct Atrifactory specific url path used for publishing")
+  val debianArtifactoryPublish = TaskKey[Unit]("debian-artifactory-publish", "Publish debian package to Artifactory")
+}
+
+object DebianArtifactoryDeployPlugin extends AutoPlugin {
+
+  override def requires = DebianPlugin
+
+  object autoImport extends DebianArtifactoryDeployKeys
+
+  import autoImport._
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/nepomukseiler/sbt-native-packager/file/1938988902/issues/source?bid=2040925&fileBranchId=3104082#l34)